extensions/azuremonitor/src/contextProvider.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+
+import * as types from './types';
+
+export enum BuiltInCommands {
+	SetContext = 'setContext',
+}
+
+export enum ContextKeys {
+	ISCLOUD = 'azuremonitor:iscloud',
+	EDITIONID = 'azuremonitor:engineedition',
+	ISCLUSTER = 'azuremonitor:iscluster',
+	SERVERMAJORVERSION = 'azuremonitor:servermajorversion'
+}
+
+const isCloudEditions = [
+	azdata.DatabaseEngineEdition.SqlDatabase,
+	azdata.DatabaseEngineEdition.SqlDataWarehouse,
+	azdata.DatabaseEngineEdition.SqlOnDemand
+];
+
+export function setCommandContext(key: ContextKeys | string, value: any) {
+	return vscode.commands.executeCommand(BuiltInCommands.SetContext, key, value);
+}
+
+export default class ContextProvider {
+	private _disposables = new Array<vscode.Disposable>();
+
+	constructor() {
+		this._disposables.push(azdata.workspace.onDidOpenDashboard(this.onDashboardOpen, this));
+		this._disposables.push(azdata.workspace.onDidChangeToDashboard(this.onDashboardOpen, this));
+	}
+
+	public onDashboardOpen(e: azdata.DashboardDocument): void {
+		let iscloud: boolean = false;
+		let edition: number | undefined;
+		let isCluster: boolean = false; // TODO: Do we even need this for AzureMonitor
```

Comment:
You shouldn't - cluster is for BDC which isn't something your stuff should care about